homeassistant/components/stream/__init__.py

```diff
@@ -0,0 +1,149 @@
+"""
+Provide functionality to stream video source.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/stream/
+"""
+import logging
+import threading
+
+import voluptuous as vol
+
+from homeassistant.auth.util import generate_secret
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.loader import bind_hass
+
+from .const import (
+    DOMAIN, CONF_KEEPALIVE, ATTR_STREAMS, ATTR_ENDPOINTS)
+from .worker import stream_worker
+from .hls import async_setup_hls
+
+REQUIREMENTS = ['av==6.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_KEEPALIVE, default=False): cv.boolean,
```

Comment:
I don't think that this is something that should be configurable for all streams. Shouldn't we make this something that is configured _per_ stream?

Reply:
I'm not sure how to do that if the streams are generated dynamically in the service calls (unless we add an option there).  As a user, I would probably want "quick load" on all my streams, or none.  Not sure if I'd want to filter down to individual streams.

Reply:
Isn't this something that we can expose in the media player play stream command ?

Reply:
Yes, but HLS is eventually going to be used for the frontend as well, so how would it get configured for users that don't implement `play_stream` in any automations?

Reply:
Can be a separate PR.

Reply:
You bet it is 😉

Reply:
Removed from global config and added it to service for now.